eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/ChannelRelay.scala

```diff
@@ -118,60 +123,63 @@ class ChannelRelay private(nodeParams: NodeParams,
 
   def relay(previousFailures: Seq[PreviouslyTried]): Behavior[Command] = {
     Behaviors.receiveMessagePartial {
-      case DoRelay =>
+      case WrappedConfidence(confidence) =>
         if (previousFailures.isEmpty) {
           context.log.info("relaying htlc #{} from channelId={} to requestedShortChannelId={} nextNode={}", r.add.id, r.add.channelId, r.payload.outgoingChannelId, nextNodeId_opt.getOrElse(""))
         }
         context.log.debug("attempting relay previousAttempts={}", previousFailures.size)
-        handleRelay(previousFailures) match {
+        handleRelay(previousFailures, confidence) match {
           case RelayFailure(cmdFail) =>
             Metrics.recordPaymentRelayFailed(Tags.FailureType(cmdFail), Tags.RelayType.Channel)
             context.log.info("rejecting htlc reason={}", cmdFail.reason)
-            safeSendAndStop(r.add.channelId, cmdFail)
+            safeSendAndStop(r.add.channelId, cmdFail, confidence, None)
           case RelaySuccess(selectedChannelId, cmdAdd) =>
             context.log.info("forwarding htlc #{} from channelId={} to channelId={}", r.add.id, r.add.channelId, selectedChannelId)
             register ! Register.Forward(forwardFailureAdapter, selectedChannelId, cmdAdd)
-            waitForAddResponse(selectedChannelId, previousFailures)
+            waitForAddResponse(selectedChannelId, previousFailures, confidence)
         }
     }
   }
 
-  def waitForAddResponse(selectedChannelId: ByteVector32, previousFailures: Seq[PreviouslyTried]): Behavior[Command] =
+  def waitForAddResponse(selectedChannelId: ByteVector32, previousFailures: Seq[PreviouslyTried], confidence: Double): Behavior[Command] =
```

Comment:
I think it's unnecessary to add the `confidence` field to every function inside channel relay. If I understand correctly, we will check the reputation of the incoming node *once*, before trying the outgoing channels, and won't modify that value during the relay retries, right? If that's the case, we should obtain it at the very beginning of this actor (directly in the `apply`) and then put `confidence` in the parameters of the `class ChannelRelay private`.